lib/async_event_emitter.js

```diff
@@ -124,18 +123,20 @@ class AsyncEventEmitter {
   async emit (eventName, ...args) {
     const indexesToRemove = []
 
-    await PI.forEachSeries(this.listeners, async (l, i) => {
+    for (const i in this.listeners) {
+      const listener = this.listeners[i]
+
       if (
-        (_isRegExp(l.matcher) && l.matcher.test(eventName)) ||
-        (l.matcher === eventName)
+        (_isRegExp(listener.matcher) && listener.matcher.test(eventName)) ||
+        (listener.matcher === eventName)
       ) {
-        await l.cb(...args)
+        await listener.cb(...args)
 
-        if (l.type === 'once') {
+        if (listener.type === 'once') {
           indexesToRemove.push(i)
         }
       }
-    })
+    }
 
     indexesToRemove.forEach(i => this.listeners.splice(i, 1))
```

Comment:
I think this is a bug.. would have to check properly though.. Might not work if `indexesToRemove` has more than one indices to splice.. since splice replaces the previous indexes